pandas/core/reshape/tile.py

```diff
@@ -527,7 +527,12 @@ def _convert_bin_to_numeric_type(bins, dtype):
             raise ValueError("bins must be of timedelta64 dtype")
     elif is_datetime64_dtype(dtype) or is_datetime64tz_dtype(dtype):
         if bins_dtype in ["datetime", "datetime64"]:
-            bins = to_datetime(bins).view(np.int64)
+            bins = to_datetime(bins)
+            if is_datetime64_dtype(bins):
+                # As of 2.0, to_datetime may give non-nano, so we need to convert
+                #  here until the rest of this file recognizes non-nano
+                bins = bins.astype("datetime64[ns]", copy=False)
```

Comment:
is there any test that would hit the `else` branch? I think now `is_datetime64_dtype(bins)` is always `True` in the test suite

Reply:
no extant test, but im not sure its impossible to reach

Reply:
perhaps raise an AssertionError in the `else` block? the it won't show up as uncovered

Reply:
im not concerned about it showing as uncovered, and don't want to break potentially-working code in the wild.

will need to revisit this eventually to handle non-nano without the cast anyway